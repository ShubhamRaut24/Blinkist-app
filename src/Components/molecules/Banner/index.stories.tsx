import BannerComponent from "./index";
import {ComponentStory, ComponentMeta} from '@storybook/react';
export default {
  title: "Molecules/Banner",
  component: BannerComponent,
}as ComponentMeta<typeof BannerComponent>;

const Banner:ComponentStory<typeof BannerComponent>= (args:any) => <BannerComponent {...args} />;

export const BannerComp = Banner.bind({});
BannerComp.args = {
//   src: "https://images.blinkist.com/images/books/608a9c296cee070007228a21/1_1/470.jpg",
//   alt: "Not Found",
//   width: "200px",
//   height: "200px",
};