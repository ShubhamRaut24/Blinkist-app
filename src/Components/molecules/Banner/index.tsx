import Image from '../../atoms/Images/index';
import Typography from '../../atoms/Typography/index';
// import Bannerimage from "./BannerImg.png";
import {Box,} from '@mui/material';



const BannerComponent = ({heading, discription, img}:any) =>{
    return (
        <Box sx={{
                backgroundColor : '#F1F6F4',
                width: 'auto',
                display: 'flex',
                padding: '4.93%',
                boxSizing: 'border-box',
                justifyContent: 'center'
            }}>

            <Box sx={{display: 'flex', flexDirection: 'column', justifyContent: 'center'}}>
                <Typography variant1="h4" sx={{fontWeight: 'bold', width: '300px'}}>
                    {heading}
                </Typography>
                <Typography  sx={{ fontWeight: 'bold', color: '#6D787E'}}>
                    {discription}
                </Typography>
            </Box>
            <Image 
                src={img}
            />
        </Box>
    );
}

export default BannerComponent;