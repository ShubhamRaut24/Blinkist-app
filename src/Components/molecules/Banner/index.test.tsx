
import { render, screen } from "@testing-library/react";
import Banner from ".";


test('Banner is Headings', () => {
    render(<Banner label='button' heading='Heading' discription='Discription'/>);
    const linkElement = screen.getByTestId('heading');
    expect(linkElement.textContent).toBe('Heading');
  });
  
  test('Bannners discriptions', () => {
    render(<Banner label='button' heading='Heading' discription='Discription'/>);
    const linkElement = screen.getByTestId('discription');
    expect(linkElement.textContent).toBe('Discription');
  });