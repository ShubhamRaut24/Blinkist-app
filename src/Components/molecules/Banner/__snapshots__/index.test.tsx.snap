// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner 1`] = `
<ForwardRef(Box)
  sx={
    Object {
      "backgroundColor": "#F1F6F4",
      "boxSizing": "border-box",
      "display": "flex",
      "justifyContent": "center",
      "padding": "4.93%",
      "width": "auto",
    }
  }
>
  <ForwardRef(Box)
    sx={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <TypographyComponents
      sx={
        Object {
          "fontWeight": "bold",
          "width": "300px",
        }
      }
      variant1="h4"
    >
      Explore Books on entrepreneurship
    </TypographyComponents>
    <TypographyComponents
      sx={
        Object {
          "color": "#6D787E",
          "fontWeight": "bold",
        }
      }
    >
      Everything you need to know about thriving on a shoestring budget, making your first million, and hiring right from the start.
    </TypographyComponents>
  </ForwardRef(Box)>
  <Image
    src="/assets/banner.png"
  />
</ForwardRef(Box)>
`;
